### PR TITLE
fix edge case duration roundings

### PR DESCRIPTION
**Pull request by JonRowe**
Based on the work of @fabersky but with @myronmarston's spec. Supersedes and fixes #2187.

### CODE REVIEW

File: lib/rspec/core/formatters/helpers.rb [hunk 1]
```diff
@@ -30,8 +30,8 @@ def self.format_duration(duration)
                       end
 
           if duration > 60
-            minutes = (duration.to_i / 60).to_i
-            seconds = duration - minutes * 60
+            minutes = (duration.round / 60).to_i
+            seconds = (duration - minutes * 60).abs
```
Comment (myronmarston) [inline lib/rspec/core/formatters/helpers.rb]: Why is the `abs` needed?  I'm trying to imagine a case where the expression in parens returns a negative value.

Comment (JonRowe) [inline lib/rspec/core/formatters/helpers.rb]: `359.999999999999 - 6*60` is negative

Comment (myronmarston) [inline lib/rspec/core/formatters/helpers.rb]: But in that case isn't it just rounded off to 0?  When I see `abs` it makes me think that we're able to get values like `-7` and then the `abs` flips it to `7`...which doesn't make much sense.

Can we just use `.round` instead?  That would be less confusing if it works.

**Comment by JonRowe** [comment]
@myronmarston I removed the `abs` in favour of forcing the rounding in the string formatting, without doing that `sprintf` treats it as `-0`

**Comment by myronmarston** [comment]
LGTM apart from the Travis failure.

**Comment by JonRowe** [comment]
So it turns out `1.8.7` can't round to `n` places, sad times, instead I'm using a guard condition to return `'0'` for the edge case less than 0 argument.

**Comment by JonRowe** [comment]
Green, final review @myronmarston ?

**Comment by myronmarston** [comment]
LGTM

**Pull request merged**